BigBaseV2/src/backend/looped/self/bulletproof.cpp

```diff
@@ -0,0 +1,13 @@
+#include "backend/looped/looped.hpp"
+#include "natives.hpp"
+
+namespace big
+{
+	void looped::self_bulletproof()
+	{
+		if (g->self.bulletproof) {
+			ENTITY::SET_ENTITY_HEALTH(self::ped, PED::GET_PED_MAX_HEALTH(self::ped), 0);
+			PED::SET_PED_ARMOUR(self::ped, PLAYER::GET_PLAYER_MAX_ARMOUR(self::id));
+		}
+	}
```

Comment:
There's a better method than using natives in a loop, if your HP is drained fast enough you'll most likely still die.

```cpp
enum class eEntityProofs
{
	BULLET = 1 << 4,
	FIRE = 1 << 5,
	COLLISION = 1 << 6,
	MELEE = 1 << 7,
	GOD = 1 << 8,
	EXPLOSION = 1 << 11,
	STEAM = 1 << 15,
	DROWN = 1 << 16
};

g_local_player->m_damage_bits |= static_cast<int>(eEntityProofs::BULLET);
```
This code achieves the same or works even better while not using any natives at all.

Don't forget to check if `g_local_player != nullptr`

Reply:
There is also a waterproof here
`waterproof = 1 << 24`

Reply:
I'm curious what's the different between ENTITY::SET_ENTITY_INVINCIBLE and the god mode mask?

Reply:
> add a waterproof `waterproof = 1 << 24`

I will add this to the file, thanks.